### PR TITLE
Test/add http test routes

### PR DESCRIPTION
**Pull request by danyecls**

### CODE REVIEW

**Review by henrybarreto — APPROVED** [review]

File: api/routes/device_test.go [hunk 1]
```diff
@@ -32,17 +91,648 @@ func TestUpdatePendingStatus(t *testing.T) {
...
+			requiredMocks: func() {
+				mock.On("LookupDevice", ctx, "123", "dispositivo1").Return(&models.Device{}, nil)
+			},
+			expectedErr: nil,
+		},
+		{
+			title: "Lookup a device not found",
+			request: `{
+				"domain": "1234",
+				"name": "dispositivo1",
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/device_test.go]: corrected for english.

File: api/routes/nsadm_test.go [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+			apictx := gateway.NewContext(mock, echoContext)
+			err := h.CreateNamespace(*apictx)
+
+			assert.Equal(t, tc.expectedErr, err)
+		})
+	}
+
+	mock.AssertExpectations(t)
+}
+func TestGetNamespace(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/nsadm_test.go:88]: add new line.

File: api/routes/nsadm_test.go [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+			apictx := gateway.NewContext(mock, echoContext)
+			err := h.GetNamespaceList(*apictx)
+
+			assert.Equal(t, tc.expectedErr, err)
+		})
+	}
+
+	mock.AssertExpectations(t)
+}
+func TestDeleteNamespace(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/nsadm_test.go:210]: add new line

File: api/routes/nsadm_test.go [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+			apictx := gateway.NewContext(mock, echoContext)
+			err := h.GetSessionRecord(*apictx)
+
+			assert.Equal(t, tc.expectedErr, err)
+		})
+	}
+
+	mock.AssertExpectations(t)
+}
+func TestEditNamespace(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/nsadm_test.go:320]: add new line

File: api/routes/nsadm_test.go [hunk 5]
```diff
@@ -0,0 +1,379 @@
...
+			apictx := gateway.NewContext(mock, echoContext)
+
+			err := h.GetNamespace(*apictx)
+
+			assert.Equal(t, tc.expectedErr, err)
+		})
+	}
+
+	mock.AssertExpectations(t)
+}
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/nsadm_test.go:135]: add new line

File: api/routes/nsadm_test.go [hunk 6]
```diff
@@ -0,0 +1,379 @@
...
+
+			err := h.DeleteNamespace(*apictx)
+
+			assert.Equal(t, tc.expectedErr, err)
+		})
+	}
+
+	mock.AssertExpectations(t)
+}
+func TestGetSessionRecord(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/nsadm_test.go:275]: add new line

File: api/routes/device_test.go [hunk 7]
```diff
@@ -32,17 +91,647 @@ func TestUpdatePendingStatus(t *testing.T) {
...
+			uid: `{
+				"uid": "123"
+			}`,
+			requiredMocks: func() {
+				mock.On("DeleteDevice", ctx, models.UID("123"), "").Return(nil)
+			},
+			expectedErr: nil,
+		},
+		{
+			title: "Delete device nout found",
```
Comment (henrybarreto) [inline api/routes/device_test.go:134]:
```suggestion
			title: "Delete device not found",
```

File: api/routes/device_test.go [hunk 8]
```diff
@@ -32,17 +91,647 @@ func TestUpdatePendingStatus(t *testing.T) {
...
+			title:             "Rename device Success",
+			RenamePayloadMock: `{"uid":"123", "name":"New Device Name"}`,
+			tenant:            "tenant-id",
+			requiredMocks: func() {
+				mock.On("RenameDevice", ctx, models.UID("123"), "New Device Name", "tenant-id").Return(nil)
+			},
+			expectedErr: nil,
+		},
+		{
+			title:             "Rename device nout found",
```
Comment (henrybarreto) [inline api/routes/device_test.go:198]: Same here

Comment (danyecls) [reply api/routes/device_test.go]: corrected writing

File: api/routes/device_test.go [hunk 9]
```diff
@@ -32,17 +91,647 @@ func TestUpdatePendingStatus(t *testing.T) {
...
+
 		mock.On("UpdatePendingStatus", ctx, models.UID("123"), models.DeviceStatus("pending"), "").Return(svc.ErrDeviceLimit)
 
 		apictx := gateway.NewContext(mock, echoContext)
-
 		output := h.UpdatePendingStatus(*apictx)
 		err, ok := output.(errors.Error)
-		assert.True(t, ok)
 
+		assert.True(t, ok)
```
Comment (otavio) [inline api/routes/device_test.go:105]: The assert should be in the previous code block, as it ensures the casting has succeeded.

File: api/routes/device_test.go [hunk 10]
```diff
@@ -14,11 +18,66 @@ import (
...
+		title         string
+		uid           string
+		requiredMocks func()
+		expectedErr   error
+	}{
+		{
+			title: "Get device success",
+			uid: `{
+				"uid": "123"
+			}`,
```
Comment (otavio) [inline api/routes/device_test.go:44]:
```suggestion
			uid: `{ "uid": "123" }`,
```

Comment (otavio) [reply api/routes/device_test.go:44]: The same code style could be applied in all tests as it reduces code size.

**Review by otavio — CHANGES_REQUESTED** [review]

**Review by gustavosbarreto — APPROVED** [review]

File: api/routes/device_test.go [hunk 11]
```diff
@@ -1,48 +1,729 @@
...
+
+	cases := []struct {
+		title           string
+		uid             string
+		requiredMocks   func()
+		expectedSession *models.Device
+		expectedStatus  int
+	}{
+		{
+			title: "Get device success",
```
Comment (otavio) [inline api/routes/device_test.go:39]:
```suggestion
			title: "returns Ok for a existing device",
```

File: api/routes/device_test.go [hunk 12]
```diff
@@ -1,48 +1,729 @@
...
+			uid:   "123",
+			requiredMocks: func() {
+				mock.On("GetDevice", gomock.Anything, models.UID("123")).Return(&models.Device{}, nil)
+			},
+			expectedSession: &models.Device{},
+			expectedStatus:  http.StatusOK,
+		},
+		{
+			title: "Get device does not exist",
```
Comment (otavio) [inline api/routes/device_test.go:48]:
```suggestion
			title: "returns Not Found for a non-existing device",
```

**Review by otavio — CHANGES_REQUESTED** [review]

File: api/routes/tags_test.go [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/labstack/echo/v4"
+	"github.com/shellhub-io/shellhub/api/pkg/echo/handlers"
+
+	"github.com/shellhub-io/shellhub/api/pkg/guard"
+	//svc "github.com/shellhub-io/shellhub/api/services"
```
Comment (henrybarreto) [inline api/routes/tags_test.go:11]: Can you remove this comment?

File: api/routes/user_test.go [hunk 14]
```diff
@@ -0,0 +1,144 @@
+package routes
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
```
Comment (henrybarreto) [inline api/routes/user_test.go:8]: Spaces occur here too

File: api/routes/device_test.go (hunk 9 above)
Comment (danyecls) [reply api/routes/device_test.go]: resolved during refactoring

File: api/routes/device_test.go (hunk 10 above)
Comment (danyecls) [reply api/routes/device_test.go]: resolved during refactoring

File: api/routes/device_test.go (hunk 11 above)
Comment (danyecls) [reply api/routes/device_test.go]: ok

File: api/routes/device_test.go (hunk 12 above)
Comment (danyecls) [reply api/routes/device_test.go]: ok

File: api/routes/device_test.go [hunk 15]
```diff
@@ -36,7 +36,7 @@ func TestGetDevice(t *testing.T) {
 		expectedStatus  int
 	}{
 		{
-			title: "Get device success",
+			title: "returns ok for a existing device",
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/device_test.go]: ok

File: api/routes/device_test.go [hunk 16]
```diff
@@ -453,15 +453,15 @@ func TestHeartbeatDevice(t *testing.T) {
 		expectedStatus int
 	}{
 		{
-			title: "Heart beat success",
+			title: "returns ok for heart beat of a existing device",
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/device_test.go]: ok

**Review by henrybarreto — DISMISSED** [review]
It seems okay to me. I would just like to remove those spaces on import :D

File: api/routes/session_test.go [hunk 17]
```diff
@@ -0,0 +1,271 @@
...
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+	"testing"
+
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/session_test.go:11]: ok

File: api/routes/tags_test.go (hunk 13 above)
Comment (danyecls) [reply api/routes/tags_test.go]: ok

**Review by otavio — CHANGES_REQUESTED** [review]

File: api/routes/device_test.go [hunk 18]
```diff
@@ -1,48 +1,729 @@
...
+			if err := json.NewDecoder(rec.Result().Body).Decode(&session); err != nil {
+				assert.ErrorIs(t, io.EOF, err)
+			}
+
+			assert.Equal(t, tc.expectedSession, session)
+		})
+	}
+}
+
+func TestOfflineDevice(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/device_test.go:339]: ok

File: api/routes/session_test.go [hunk 19]
```diff
@@ -0,0 +1,271 @@
...
+	"github.com/shellhub-io/shellhub/api/pkg/guard"
+	"github.com/shellhub-io/shellhub/api/services/mocks"
+	"github.com/shellhub-io/shellhub/pkg/api/paginator"
+	"github.com/shellhub-io/shellhub/pkg/api/requests"
+	"github.com/shellhub-io/shellhub/pkg/models"
+	"github.com/stretchr/testify/assert"
+	gomock "github.com/stretchr/testify/mock"
+)
+
+func TestGetSessionList(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/session_test.go:27]: ok

**Review by otavio — CHANGES_REQUESTED** [review]

File: api/routes/healthcheck_test.go [hunk 20]
```diff
@@ -0,0 +1,44 @@
...
+	cases := []struct {
+		title         string
+		requiredMocks func()
+		expectedErr   error
+	}{
+		{
+			title:       "returns Ok for evaluate health",
+			expectedErr: nil,
+		},
+	}
```
[the remark this answers is not in the document]
Comment (danyecls) [reply api/routes/healthcheck_test.go:28]: add newline

**Review by otavio — DISMISSED** [review]

**Pull request merged**